src/ui/public/vis/map/service_settings.js

```diff
@@ -139,8 +142,16 @@ uiModules.get('kibana')
       async _getManifest(manifestUrl) {
         return $http({
           url: extendUrl(manifestUrl, { query: this._queryParams }),
-          method: 'GET'
-        });
+          method: 'GET',
+          timeout: 4000
+        }).then(resp => resp.data)
+          .catch(resp => {
+            toastNotifications.addDanger(
```

Comment:
very cool, nice!

Reply:
I liked it here too, but actually had to move it to account for the fact that we really only want to show a warning if it's the default, which isn't the case of a TMS is supplied.